aws_lambda_powertools/utilities/data_classes/api_gateway_authorizer_event.py

```diff
@@ -0,0 +1,300 @@
+    def get_type(self) -> str:
+        return self["type"]
+
+    @property
+    def authorization_token(self) -> str:
+        return self["authorizationToken"]
+
+    @property
+    def method_arn(self) -> str:
+        """ARN of the incoming method request and is populated by API Gateway in accordance with the Lambda authorizer
+        configuration"""
+        return self["methodArn"]
+
+    @property
+    def parsed_arn(self) -> APIGatewayRouteArn:
+        """Convenient property to return a parsed api gateway method arn"""
+        return parse_api_gateway_arn(self.method_arn)
+
+
+class APIGatewayAuthorizerRequestEvent(DictWrapper):
+    """API Gateway Authorizer Request Event Format 1.0
+
+    Documentation:
+    -------------
+    - https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html
+    - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-lambda-authorizer.html
+    """
+
+    @property
+    def version(self) -> str:
+        return self["version"]
+
+    @property
+    def get_type(self) -> str:
+        return self["type"]
+
+    @property
+    def method_arn(self) -> str:
+        return self["methodArn"]
+
+    @property
+    def parsed_arn(self) -> APIGatewayRouteArn:
+        return parse_api_gateway_arn(self.method_arn)
+
+    @property
+    def identity_source(self) -> str:
+        return self["identitySource"]
+
+    @property
+    def authorization_token(self) -> str:
+        return self["authorizationToken"]
+
+    @property
+    def resource(self) -> str:
+        return self["resource"]
+
+    @property
+    def path(self) -> str:
+        return self["path"]
+
+    @property
+    def http_method(self) -> str:
+        return self["httpMethod"]
+
+    @property
+    def headers(self) -> Dict[str, str]:
+        return self["headers"]
+
+    @property
+    def query_string_parameters(self) -> Dict[str, str]:
+        return self["queryStringParameters"]
+
+    @property
+    def path_parameters(self) -> Dict[str, str]:
+        return self["pathParameters"]
+
+    @property
+    def stage_variables(self) -> Dict[str, str]:
+        return self["stageVariables"]
+
+    @property
+    def request_context(self) -> BaseRequestContext:
+        return BaseRequestContext(self._data)
+
+    def get_header_value(
+        self, name: str, default_value: Optional[str] = None, case_sensitive: Optional[bool] = False
+    ) -> Optional[str]:
+        """Get header value by name
+
+        Parameters
+        ----------
+        name: str
+            Header name
+        default_value: str, optional
+            Default value if no value was found by name
+        case_sensitive: bool
+            Whether to use a case sensitive look up
+        Returns
+        -------
+        str, optional
+            Header value
+        """
+        return get_header_value(self.headers, name, default_value, case_sensitive)
+
+
+class APIGatewayAuthorizerSimpleResponse:
```

Comment:
```suggestion
class APIGatewayAuthorizerV2Response:
```

This makes it consistent with what you chose previously: `APIGatewayAuthorizerV2Event`. It's also easier to differentiate when `APIGatewayAuthorizerV1Response` is implemented

Reply:
@heitorlessa  but confusingly the V2 format events, also support the V1 format response (For the Auth) :) (T